src/lxc/console.c

```diff
@@ -196,6 +196,15 @@ static int lxc_console_cb_con(int fd, uint32_t events, void *data,
 
 		if (console->peer >= 0)
 			w = lxc_write_nointr(console->peer, buf, r);
+
+		if (console->log_size > 0) {
```

Comment:
I *think* this is right, but just to make sure - is there any case where a console reader or writer will be in a different thread?

Reply:
@hallyn, I don't think so. We need to go through the container's command socket to request an operation on the console ringbuffer to be performed. The command socket handler and console handler will always be in the same thread since they are setup together on container in start in the epoll loop.

Reply:
My point being that the lock needed for an operation on the ringbuffer will always be taken by the monitor.